app/components/sidebars/main/channels_list/filtered_list/index.js

```diff
@@ -98,23 +99,25 @@ const getGroupChannelMemberDetails = createSelector(
 
 function mapStateToProps(state) {
     const {currentUserId} = state.entities.users;
+    const config = getConfig(state);
 
     const profiles = getUsers(state);
     let teamProfiles = {};
-    const restrictDms = getConfig(state).RestrictDirectMessage !== General.RESTRICT_DIRECT_MESSAGE_ANY;
+    const restrictDms = config.RestrictDirectMessage !== General.RESTRICT_DIRECT_MESSAGE_ANY;
     if (restrictDms) {
         teamProfiles = getTeamProfiles(state);
     }
 
     const searchOrder = Config.DrawerSearchOrder ? Config.DrawerSearchOrder : DEFAULT_SEARCH_ORDER;
+    const viewArchivedChannels = config.ExperimentalViewArchivedChannels === 'true';
 
     return {
         channels: getChannelsWithUnreadSection(state),
         currentChannel: getCurrentChannel(state),
         currentTeam: getCurrentTeam(state),
         currentUserId,
         otherChannels: getOtherChannels(state, false),
-        archivedChannels: getArchivedChannels(state),
+        archivedChannels: viewArchivedChannels ? getArchivedChannels(state) : emptyArray,
```

Comment:
Maybe we can just directly place `[]` instead of declaring an `emptyArray = []`.

Reply:
Passing `[]` every time is a nee array every time and will cause uneeded re-renders